### PR TITLE
add the JSONParser

### PR DESCRIPTION
**Pull request by JavierJia**
Add the JSON parser that can convert the JSON request to Query object. 

@MoniMoledo @aishwaryakapse 

related PR #128 
related issue #30

### CODE REVIEW

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala [hunk 1]
```diff
@@ -6,12 +6,35 @@ import org.specs2.mutable.Specification
...
-    "parse the by (spatial, time) count" in {
-      //val actualQuery = parser.parse(stateHourCountJSON)
-      //val expectQuery = new Query(schema.dataset, Seq.empty, Seq.empty, Seq.empty,
-      //Some(GroupStatement(Seq(byState, byHour), Seq(aggrCount))), None)
-      //actualQuery must_== expectQuery
+    "parse the hourly count request" in {
+      val actualQuery = parser.parse(hourCountJSON)
+      val expectQuery = Query(schema.dataset, Seq.empty, Seq.empty, Seq.empty,
+                              Some(GroupStatement(Seq(byHour), Seq(aggrCount))), None)
+      actualQuery.groups must_== expectQuery.groups
```
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: Why did you verify the groups match in addition to verify if the whole query matches ?

File: zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala [hunk 2]
```diff
@@ -6,12 +6,35 @@ import org.specs2.mutable.Specification
...
+      val expectQuery = Query(schema.dataset, Seq.empty, filter, Seq.empty, Some(group), None)
+      actualQuery must_== expectQuery
+    }
+    "parse the by topK hashtag request" in {
+      val actualQuery = parser.parse(topKHashTagJSON)
+      val filter = Seq(stateFilter, timeFilter, textFilter)
+      val group = GroupStatement(Seq(byTag), Seq(aggrCount))
+      val expectQuery = Query(schema.dataset, Seq.empty, filter, Seq(unnestHashTag), Some(group), Some(selectTop10Tag))
+      actualQuery must_== expectQuery
+      ok
```
Comment (MoniMoledo) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: What is "ok" doing here ?

Comment (JavierJia) [inline zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala]: it's a place marker before I fill up the test case. removed.

**Comment by MoniMoledo** [comment]
:+1:

**Pull request merged**